LambdaEngine/Source/Memory/API/StackAllocator.cpp

```diff
@@ -0,0 +1,81 @@
+#include "Memory/API/StackAllocator.h"
+
+namespace LambdaEngine
+{
+	/*
+	* StackAllocator
+	*/
+
+	StackAllocator::StackAllocator(uint32 sizePerArena)
+		: m_ArenaIndex(0)
+		, m_SizePerArena(sizePerArena)
+		, m_pCurrentArena(nullptr)
+		, m_Arenas()
+	{
+		m_pCurrentArena = &m_Arenas.EmplaceBack(m_SizePerArena);
+	}
+
+	void* StackAllocator::Push(uint32 size)
+	{
+		VALIDATE(m_pCurrentArena != nullptr);
+		
+		// Do we need a new arena or can we take next one?
+		if (!m_pCurrentArena->CanAllocate(size))
+		{
+			m_ArenaIndex++;
+			if (m_Arenas.GetSize() <= m_ArenaIndex)
+			{
+				m_pCurrentArena = &m_Arenas.EmplaceBack(m_SizePerArena);
+			}
+			else
+			{
+				m_pCurrentArena = &m_Arenas[m_ArenaIndex];
+				m_pCurrentArena->Reset();
+			}
+
+			VALIDATE(m_pCurrentArena != nullptr);
+
+			// In case we still cannot allocate
+			if (!m_pCurrentArena->CanAllocate(size))
+			{
+				return nullptr;
+			}
+		}
+
+		// Allocate
+		return m_pCurrentArena->Allocate(size);
+	}
+
+	void StackAllocator::Pop(uint32 size)
+	{
+		VALIDATE(m_pCurrentArena != nullptr);
+
+		if (!m_pCurrentArena->CanPop(size))
+		{
+			if (m_ArenaIndex > 0)
+			{
+				m_ArenaIndex--;
+				m_pCurrentArena = &m_Arenas[m_ArenaIndex];
+				
+				VALIDATE(m_pCurrentArena != nullptr);
+			}
+			else
+			{
+				return;
+			}
```

Comment:
This is removable since `MemoryArena::Pop` doesn't do anything if it can't pop. If you do remove it, you can do `if (expression0 && expression1)` instead of nesting if-statements.